plugins/database/src/db_plugin.cpp

```diff
@@ -8946,14 +8987,19 @@ irods::error db_set_avu_metadata_op(
              * alternative, but this is a definite TODO: Implement AVUMetadata locks.
              */
             status = chlDeleteAVUMetadata( _ctx.comm(), 1, _type, _name, _attribute, "%",
-                                           "%", 1 );
+                                           "%", 1, _cond_input );
         }
+
         if ( status != 0 ) {
             _rollback( "chlSetAVUMetadata" );
             return ERROR( status, "delete avu metadata failed" );
         }
-        status = chlAddAVUMetadata( _ctx.comm(), 0, _type, _name, _attribute,
-                                    _new_value, _new_unit );
+
+        // TODO This function used to be executed with _admin_mode set to zero.
+        // Are the semantics identical when _cond_input holds ADMIN_KW?
+        status = chlAddAVUMetadata( _ctx.comm(), _type, _name, _attribute,
+                                    _new_value, _new_unit, _cond_input );
```

Comment:
remove TODO comment.

if the admin keyword is set, all calls that deal with permissions should honor it.